src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java

```diff
@@ -18,13 +19,13 @@
 
 public class PentahoAsyncReportExecution implements IAsyncReportExecution<InputStream> {
 
-  private final SimpleReportingComponent reportComponent;
-  private final AsyncJobFileStagingHandler handler;
-  private final String url;
-  private final String auditId;
-  private final IPentahoSession safeSession;
+  protected final SimpleReportingComponent reportComponent;
+  protected final AsyncJobFileStagingHandler handler;
+  protected final String url;
+  protected final String auditId;
+  protected final IPentahoSession safeSession;
 
-  private AsyncReportStatusListener listener;
+  protected AsyncReportStatusListener listener;
```

Comment:
I'm fine with "protected final" members being accessed directly, as there is no harm anyone can do to the internal state of the class. But for the "listener" please use getters/setters to control access. If you only need read access to the listener, then make sure you only provide a getter.

Also: I'd like to see the context of these changes so I'll hold off merging this change until there is some PIR code showing how these protected members are used. As you are working in separate branches for each feature you work on, that should not block you from developing the features.